app/models/user.rb

```diff
@@ -1,11 +1,18 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
 
-  after_commit :create_shopping_cart
+  after_create :create_shopping_cart
+  after_save :send_welcome_mail
```

Comment:
this will send email on every update

Reply:
Changed to : after_create :send_welcome_mail

Reply:
Changed to : after_create :send_welcome_mail